fluxy/cli/utils_annex_plot.py

```
@@ -33,99 +31,123 @@ def get_species_specific_settings(
-        if len(inv_val) == 1:
-            output[str(time)] = [f"{inv_val[0]:.{n_digits}f}", paris_val]
+    if not sector:
+        if res.sector.unique().size!=1:
+            raise ValueError(
+                f"`sector` parameter should be provided when there is more than one region in `res` (currently present: {res.sector.unique()})."
+            )
+        sector = res.sector.unique()[0]
+
+    if not isinstance(species,list):
+        species = [species,]
+
+    res["time"] = pd.to_datetime(res["time"])
+    data = res[(res.country==region)
+                &(res.model.isin([model,f"inventory_{inventory_year}"]))
+                &res.species.isin(species)
+                &(res.type.isin(["posterior","inventory"]))
+                &(res.time>=table_start_date)
+                ].reset_index(drop=True)
+    
+    data["year"] = pd.to_datetime(data["time"]).dt.year.astype(str)
+    species_order = data[data.model==model].groupby("species").mean_val.mean().sort_values(ascending=False).index
+
+    rescaled_data = list()
+    for species in data.species.unique():
+        data_per_species = data[data.species==species].copy()
+
+        _, exp = np.stack(data_per_species.mean_val.apply(lambda x: np.array(f"{x:.2e}".split("e")).astype(float)
+                                                            if f"{x:.2e}"!="0.00e+00" else [np.nan,np.nan]).values).T
+        max_exp = np.nanmax(exp)
+        if max_exp<-1 and max_exp>=-4:
+            for var in ["mean_val","min_unc","max_unc"]:
+                data_per_species[var] *= 1e3
+            data_per_species["units"] = "$\\rm{GgCO}_{2}\\rm{-eq} \\cdot \\rm{yr}^{-1}$"
+            max_exp += 3
+        elif max_exp<-4:
+            for var in ["mean_val","min_unc","max_unc"]:
+                data_per_species[var] *= 1e6
+            data_per_species["units"] = "$\\rm{MgCO}_{2}\\rm{-eq} \\cdot \\rm{yr}^{-1}$"
+            max_exp += 6
```

This part of the code assumes that `country_flux_units_print = "Tg CO2-eq yr-1"` in config_annex_plot.py which might not always be the case. I don't think you need to loose the time to make it generic but maybe you can raise an error if different units are used for `country_flux_units_print`.

---

Good point.
I put a note in the description. But a correction will need to wait. The unit attribute does not mke it into the combined dataset apparently. So it will need a bit of digging to fix.
I suggest going on with this and opening an issue about it.